test/regression/targets/sc-60301.cc

```diff
@@ -0,0 +1,92 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <test/support/tdb_catch.h>
+#include <tiledb/tiledb>
+
+using namespace tiledb;
+
+static void create_array(Context& ctx, const std::string& array_uri);
+static void write_array(Context& ctx, const std::string& array_uri);
+
+TEST_CASE(
+    "SC-60301: Read data with global cell order returns fill values",
+    "[dense-reader][bug][global-cell-order][fixed][sc60301]") {
+  Context ctx;
+  std::string array_uri = "dense_global_cell_order";
+
+  // Test setup
+  create_array(ctx, array_uri);
+  write_array(ctx, array_uri);
+
+  Array array(ctx, array_uri, TILEDB_READ);
+  Subarray subarray(ctx, array);
+  subarray.set_subarray({1, 2, 1, 2});
+  std::vector<int> a_read;
+  a_read.resize(4);
+  Query query(ctx, array);
+  query.set_subarray(subarray)
+      .set_layout(TILEDB_GLOBAL_ORDER)
+      .set_data_buffer("a", a_read);
+
+  REQUIRE(query.submit() == Query::Status::COMPLETE);
+
+  REQUIRE(a_read[0] == 1);
+  REQUIRE(a_read[1] == 3);
+  REQUIRE(a_read[2] == 2);
+  REQUIRE(a_read[3] == 4);
+
+  array.close();
+}
+
+void create_array(Context& ctx, const std::string& array_uri) {
+  auto obj = Object::object(ctx, array_uri);
+  if (obj.type() != Object::Type::Invalid) {
+    Object::remove(ctx, array_uri);
+  }
+
+  Domain domain(ctx);
+  domain.add_dimension(Dimension::create<int>(ctx, "d1", {{1, 2}}, 2))
+      .add_dimension(Dimension::create<int>(ctx, "d2", {{1, 2}}, 2));
+
+  // Create the array schema with col-major cell order and tile order
+  ArraySchema schema(ctx, TILEDB_DENSE);
+  schema.set_domain(domain)
+      .set_order({{TILEDB_COL_MAJOR, TILEDB_COL_MAJOR}})
+      .add_attribute(Attribute::create<int>(ctx, "a"));
+  Array::create(ctx, array_uri, schema);
+}
+
+void write_array(Context& ctx, const std::string& array_uri) {
+  std::vector<int> data = {1, 2, 3, 4};
+  Array array(ctx, array_uri, TILEDB_WRITE);
+  Query query(ctx, array);
+  query.set_layout(TILEDB_ROW_MAJOR).set_data_buffer("a", data);
```

Comment:
I know that's a regression test supposed to repro the exact issue, but while we are here could you add a GENERATE to also check writing with other layouts than `ROW_MAJOR`?